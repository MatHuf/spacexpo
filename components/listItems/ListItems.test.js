import React from "react";
import renderer from "react-test-renderer";
import {
  HistoryListItem,
  LaunchListItem,
  CapsuleRenderItem,
  RocketRenderItem,
  ShipRenderItem
} from "./index";

const mockHistoryItem = {
  item: {
    title: "History Title",
    event_date_utc: "2000-01-01",
    details: "History Details"
  }
};

const mockLaunchItem = {
  item: {
    flight_number: "0",
    rocket: {
      rocket_name: "Rocket Name"
    },
    launch_site: {
      site_name_long: "Site Name"
    },
    detail: "Details"
  }
};

const mockCapsuleItem = {
  item: {
    name: "Capsule Name",
    crew_capacity: "2",
    first_flight: "2000-01-01",
    description: "Capsule Description"
  }
};

const mockRocketItem = {
  item: {
    rocket_name: "Rocket Name",
    first_flight: "2000-01-01",
    engines: {
      number: "5",
      type: "Fancy"
    },
    description: "Rocket Description"
  }
};

const mockShipItem = {
  item: {
    ship_id: "Ship Id",
    ship_name: "Ship Name",
    successful_landings: "1",
    attempted_landings: "1"
  }
};

describe("List Items", () => {
  it("<HistoryListItem /> renders correctly", () => {
    const tree = renderer
      .create(React.createElement(() => HistoryListItem(mockHistoryItem)))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("<LaunchListItem /> renders correctly", () => {
    const tree = renderer
      .create(React.createElement(() => LaunchListItem(mockLaunchItem)))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("<CapsuleRenderItem /> renders correctly", () => {
    const tree = renderer
      .create(React.createElement(() => CapsuleRenderItem(mockCapsuleItem)))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("<RocketRenderItem /> renders correctly", () => {
    const tree = renderer
      .create(React.createElement(() => RocketRenderItem(mockRocketItem)))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("<ShipRenderItem /> renders correctly", () => {
    const tree = renderer
      .create(React.createElement(() => ShipRenderItem(mockShipItem)))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
