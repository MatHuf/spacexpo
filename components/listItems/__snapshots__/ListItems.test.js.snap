// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Items <CapsuleRenderItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "x": -5,
        "y": -5,
      },
      "shadowOpacity": 50,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333333",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      Capsule Name
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Crew: 
      2
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      First flight: 
      2000-01-01
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "justify",
        }
      }
    >
      Capsule Description
    </Text>
  </View>
</View>
`;

exports[`List Items <HistoryListItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "x": -5,
        "y": -5,
      },
      "shadowOpacity": 50,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333333",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      History Title
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      2000-01-01
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "justify",
        }
      }
    >
      History Details
    </Text>
  </View>
</View>
`;

exports[`List Items <LaunchListItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "x": -5,
        "y": -5,
      },
      "shadowOpacity": 50,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333333",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      Launch 
      0
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Rocket Name
       | 
      Site Name
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "justify",
        }
      }
    >
      No description available
      .
    </Text>
  </View>
</View>
`;

exports[`List Items <RocketRenderItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "x": -5,
        "y": -5,
      },
      "shadowOpacity": 50,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333333",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      Rocket Name
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      First flight: 
      2000-01-01
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Engines: 
      5
      x 
      Fancy
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "justify",
        }
      }
    >
      Rocket Description
    </Text>
  </View>
</View>
`;

exports[`List Items <ShipRenderItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "display": "flex",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "x": -5,
        "y": -5,
      },
      "shadowOpacity": 50,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333333",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      Ship Id
       | 
      Ship Name
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "justify",
        }
      }
    >
      Successful landings: 
      1
      /
      1
    </Text>
  </View>
</View>
`;
